lib/rucio/db/sqla/migrate_repo/versions/1677d4d803c8_split_rse_availability_into_multiple.py

```diff
@@ -0,0 +1,127 @@
+# -*- coding: utf-8 -*-
+# Copyright European Organization for Nuclear Research (CERN) since 2012
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+""" split rse availability into multiple """
+
+import sqlalchemy as sa
+from alembic import context
+from alembic.op import add_column, drop_column, get_bind
+
+from rucio.common.utils import Availability
+from rucio.db.sqla.types import GUID
+
+# Alembic revision identifiers
+revision = "1677d4d803c8"
+down_revision = "fe1a65b176c9"
+
+
+SCHEMA = context.get_context().version_table_schema if context.get_context().version_table_schema else ""
+
+OLD_RSE = sa.sql.table(
+    "rses",
+    sa.Column("id", GUID()),
+    sa.Column("availability", sa.Integer),
+    schema=SCHEMA,
+)
+
+NEW_RSE = sa.sql.table(
+    "rses",
+    sa.Column("id", GUID()),
+    sa.Column("availability_read", sa.Boolean),
+    sa.Column("availability_write", sa.Boolean),
+    sa.Column("availability_delete", sa.Boolean),
+    schema=SCHEMA,
+)
+
+
+def upgrade():
+    """
+    Upgrade the database to this revision
+    """
+
+    conn = get_bind()
```

Comment:
I think this is missing some transaction?

Reply:
This `get_bind` lets us use the alembic connection to the database. https://alembic.sqlalchemy.org/en/latest/ops.html#alembic.operations.Operations.get_bind

https://alembic.sqlalchemy.org/en/latest/ops.html#alembic.operations.Operations.execute
This describes the usage, I think it should be fine.